File: dsa/4.sorting/88_merge_sorted_array.java
package sorting;
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int nums1Ind = m - 1;
        int nums2Ind = n - 1;
        int numsInd = m + n - 1;

        while (nums1Ind >= 0 && nums2Ind >= 0){
            if (nums2[nums2Ind] > nums1[nums1Ind]) {
                nums1[numsInd] = nums2[nums2Ind];
                nums2Ind -= 1;
            }
            else{
                nums1[numsInd] = nums1[nums1Ind];
                nums1Ind -= 1;
            }
            numsInd -= 1;
        }

        while (nums1Ind >= 0){
            nums1[numsInd] = nums1[nums1Ind];
            nums1Ind -= 1;  
            numsInd -= 1;         
        }

        while (nums2Ind >= 0){
            nums1[numsInd] = nums2[nums2Ind];
            nums2Ind -= 1; 
            numsInd -= 1;
        }
    }
}

File: dsa/4.sorting/88_Merge_Sorted_Array.py
import typing as t

class Solution:
    def merge(self, nums1: t.List[int], m: int, nums2: t.List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """

        ind1: int = m - 1
        ind2: int = n - 1
        ind: int = m + n - 1

        while ind1 >= 0 and ind2 >= 0:
            if nums1[ind1] > nums2[ind2]:
                nums1[ind] = nums1[ind1]
                ind1 -= 1
            else:
                nums1[ind] = nums2[ind2]
                ind2 -= 1
            ind -= 1
        
        while ind2 >= 0:
            nums1[ind] = nums2[ind2]
            ind2 -= 1
            ind -= 1
        